exchange/exchange.go

```diff
@@ -1573,3 +1585,19 @@ func setErrorMessageSecureMarkup(validationType string) string {
 	}
 	return ""
 }
+
+// setSeatNonBid adds SeatNonBids within bidResponse.Ext.Prebid.SeatNonBid
+func setSeatNonBid(bidResponseExt *openrtb_ext.ExtBidResponse, seatNonBids nonBids) *openrtb_ext.ExtBidResponse {
+	if len(seatNonBids.seatNonBidsMap) == 0 {
+		return bidResponseExt
+	}
+	if bidResponseExt == nil {
+		bidResponseExt = &openrtb_ext.ExtBidResponse{}
+	}
+	if bidResponseExt.Prebid == nil {
+		bidResponseExt.Prebid = &openrtb_ext.ExtResponsePrebid{}
+	}
+
+	bidResponseExt.Prebid.SeatNonBid = seatNonBids.get()
+	return bidResponseExt
```

Comment:
These lines of code aren't covered by tests, can this be updated?

Reply:
<img width="1081" alt="image" src="https://github.com/prebid/prebid-server/assets/9314645/bcff3247-bb62-4b37-a604-2046f0a56581">
Done. But unable to cover else part as its hard to produce `json.Marshal` error scenario

Reply:
Done. 
<img width="999" alt="image" src="https://github.com/prebid/prebid-server/assets/9314645/e592c586-95bb-4fa2-b10f-c920b4c85b1c">